calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -0,0 +1,324 @@
+                ({'--ip':'127.0.0.1', '--ip6':'127.0.0.1'},False),
+                ({'--ip':'127.0.0.1', '--ip6':'aa:bb::zz'},False))
+
+        # Call method under test for each test case
+        # Assert that method exits on bad input
+        for case, is_valid in cases:
+            print "Testing case %s ..." % case
+            node_validate_arguments(case)
+            if is_valid:
+                assert not m_sys_exit.called
+            else:
+                assert m_sys_exit.called
+            m_sys_exit.reset_mock()
+
+
+class TestPool(unittest.TestCase):
+
+    @patch('sys.exit',autospec=True)
+    def test_validate_arguments(self, m_sys_exit):
+        """
+        Test validate_arguments for calicoctl pool command
+        """
+        # Set up arguments
+        cases = (({'add':1,'<CIDR>':'127.a.0.1'},False),
+                 ({'add':1,'<CIDR>':'aa:bb::zz'},False),
+                 ({'add':1,'<CIDR>':'1.2.3.4'},True),
+                 ({'add':1,'<CIDR>':'1.2.3.0/24'},True),
+                 ({'add':1,'<CIDR>':'aa:bb::ff'},True))
+
+        # Call method under test for each test case
+        # Assert that method exits on bad input
+        for case, is_valid in cases:
+            print "Testing case %s ..." % case
+            pool_validate_arguments(case)
+            if is_valid:
+                assert not m_sys_exit.called
+            else:
+                assert m_sys_exit.called
+            m_sys_exit.reset_mock()
+
+
+class TestProfile(unittest.TestCase):
+
+    @patch('sys.exit',autospec=True)
+    def test_validate_arguments(self, m_sys_exit):
+        """
+        Test validate_arguments for calicoctl profile command
+        """
+        # Set up arguments
+        cases = (({'<PROFILES>':'profile-1' 'profile-2' 'profile-3'},True),
+                 ({'<PROFILE>':'profile1'},True),
+                 ({'<PROFILES>':'Profile1' 'Profile!'},False),
+                 ({'<SRCTAG>':'Tag-1','<DSTTAG>':'Tag-2'},True),
+                 ({'<SRCTAG>':'Tag~1','<DSTTAG>':'Tag~2'},False),
+                 ({'<SRCCIDR>':'127.a.0.1'},False),
+                 ({'<DSTCIDR>':'aa:bb::zz'},False),
+                 ({'<SRCCIDR>':'1.2.3.4','<DSTCIDR>':'1.2.3.4'},True),
+                 ({'<ICMPCODE>':'5'},True),
+                 ({'<ICMPTYPE>':'16'},True),
+                 ({'<ICMPCODE>':100,'<ICMPTYPE>':100},True),
+                 ({'<ICMPCODE>':4,'<ICMPTYPE>':255},False),
+                 ({},True))
+
+        # Call method under test for each test case
+        # Assert that method exits on bad input
+        for case, is_valid in cases:
+            print "Testing case %s ..." % case
+            profile_validate_arguments(case)
+            if is_valid:
+                assert not m_sys_exit.called
+            else:
+                assert m_sys_exit.called
+            m_sys_exit.reset_mock()
+
+class TestStatus(unittest.TestCase):
+
+    def setUp(self):
+        pass
```

Comment:
You can get rid of this class too, I don't think there will be any functions to test within the status.py file